framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3106,7 +3106,7 @@ public boolean validateTypeOf(Tree tree) {
             }
             break;
         default:
-            type = atypeFactory.getAnnotatedType(tree);
+        	type = atypeFactory.getDefaultedAnnotatedType(tree);
```

Comment:
Mier (topnessman) is hopefully finished with cleaning up the repo soon, then he can show the current status.
The type rule is that `@Rep` is forbidden in static contexts, e.g. in a static method/field.
Then, we had `@Any Object o = new @Rep Object();` where `@Any` is a supertype of `@Rep`.
We expect one error for the use of `Rep` in the object creation, but not a second one for the local variable or field type.